app/controllers/admin/communities_controller.rb

```diff
@@ -180,12 +180,29 @@ def test_welcome_email
   def settings
     @selected_left_navi_link = "admin_settings"
 
-    render :settings, locals: {
-             supports_escrow: escrow_payments?(@current_community),
-             delete_redirect_url: delete_redirect_url(APP_CONFIG),
-             delete_confirmation: @current_community.ident,
-             can_delete_marketplace: can_delete_marketplace?(@current_community.id)
-           }
+    locals = {
+      supports_escrow: escrow_payments?(@current_community),
+      delete_redirect_url: delete_redirect_url(APP_CONFIG),
+      delete_confirmation: @current_community.ident,
+      can_delete_marketplace: can_delete_marketplace?(@current_community.id)
+    }
+
+    # When feature flag is removed, make this pretty
+    if feature_enabled?(:location_search)
+      search_settings = MarketplaceService::API::Marketplaces.search_settings(community_id: @current_community.id).data
+
+      main_search_select_options = [:keyword, :location]
+        .map { |type|
+          [SettingsViewUtils.search_type_translation(type), type]
+        }
+
+      locals.merge!({
```

Comment:
The whole merge should be removed when we remove the feature flag. Therefore, I thought that this is more clear of what is the unnecessary part of code - maybe I missed that goal.